qt/scientific_interfaces/EnggDiffraction/RunMap.tpp

```diff
@@ -0,0 +1,94 @@
+#include <algorithm>
+
+namespace {
+
+template <typename T> void insertInOrder(const T &item, std::vector<T> &vec) {
+  vec.insert(std::upper_bound(vec.begin(), vec.end(), item), item);
+}
+}
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+template <size_t NumBanks, typename T>
+void RunMap<NumBanks, T>::add(const int runNumber, const size_t bank,
+                              const T itemToAdd) {
+  if (bank < 1 || bank > NumBanks) {
+    throw std::invalid_argument("Tried to access invalid bank: " +
+                                std::to_string(bank));
+  }
+  m_map[bank - 1][runNumber] = itemToAdd;
+}
+
+template <size_t NumBanks, typename T>
+bool RunMap<NumBanks, T>::contains(const int runNumber,
+                                   const size_t bank) const {
+  return bank > 0 && bank <= NumBanks &&
+         m_map[bank - 1].find(runNumber) != m_map[bank - 1].end();
+}
+
+template <size_t NumBanks, typename T>
+const T &RunMap<NumBanks, T>::get(const int runNumber,
+                                  const size_t bank) const {
+  if (bank < 1 || bank > NumBanks) {
+    throw std::invalid_argument("Tried to access invalid bank: " +
+                                std::to_string(bank));
+  }
```

Comment:
Would it make sense to refactor this test to a separate function? It is needed here, in `RunMap::add()` and (possibly) `RunMap::remove()`.